src/Parser/Template.cs

```diff
@@ -54,6 +57,95 @@ public Template(string template, bool isPositional, List<Literal> literals, List
       Holes = holes.ToArray();
     }
 
+    public string Render(object[] parameters)
```

Comment:
> Well or just public?

Public feels wrong, they are implementation details. Especially `Literals`, which is not all that intuitive.

**Or** the `Template` class itself is not public and then making members `public` is like making them `internal`.

> We could move it to partial/new class. 

Yeah, and even hide it behind a `#if`. ;)